### DISCUSSION TITLE
Further development of this library fork

### DISCUSSION THREAD

**Discussion by andregasser** [andregasser/bigbone#27] [Ideas]
As you might have noticed, I have detached my fork from the original authors repository. This is mainly because I do not expect pull requests to be accepted within reasonable time. In order to bring this library code base forward, I would like to further develop and release this independently. 

Based on the above, I would like to perform some basic modifications to this code base:

1. Rename the library from mastodon4j to something else, because two Java clients having the same name but different repos create confusion amongst potential users.

2. Rename the current package structure from com.sys1yagi.mastoton4j to something that reflects this project better. 

3. Since this library did not have a single release yet, we could also start releasing an initial version 1.0.0 (or maybe better a 2.0.0) and build up based on that version.

Any objections?

@bocops

**Comment by bocops** [comment]
I agree that activity on the original repository seems unlikely at least in the near future. All three items sound reasonable - regarding version number, I'd favor 2.0.0 for the inital release because of the code's history that will stay available.

Do you want to more explicitly define the scope of 2.0.0 vs. anything that should come later, so that we can focus on those aspects first?

**Reply by andregasser** [reply]
Hello @bocops , yes that makes sense. Let's define a scope for 2.0.0 milestone. I have already created some issues for 1.8.0 milestone, but will move them to 2.0.0.

**Comment by andregasser** [comment]
Here's some input for library name and package structure (will add some more in the coming hours / days):

__Requirements__
- Name should be short and easy to pronounce, easy to write

__Library Name__
Some names that came to my mind combined with some inputs from people on Mastodon (thank you!): 
- mammut4j
- mastonaut
- justodon (contributed by Pasha Finkelshteyn, @asm0dey@fosstodon.org)
- leviathan (contributed by @factotum@c.im)
- beringia (contributed by @factotum@c.im)
- mastodon-client
- jmastodon
- social4j
- jsocial
- bigbone4j (Big Bone Lick State Park, Big Bone - first bones to be collected and studied scientifically)
- bone4j
- jbone
- giant4j

__Package Structure__
We could either go with something like "com.github.andregasser.*" or with some custom domain like "tld.domainname.*".I do not really want my name to be reflected in the package structure, for various reasons:
- Should I ever decide to handover the project to someone else, they do not face the same issue we have now.
- It is too much glory on my name - it is a community effort

**Reply by bocops** [reply]
Full disclosure, I'm the one who suggested Leviathan and Beringia over on Mastodon:
- "Leviathan" is a synonym for Mastodon/Mammoth
- "Beringia" is the name of the prehistoric subcontinent between Asia and North America; Mammoths existed there

I'm immediately drawn to "bigbone" on that list. It is short and easy, and even a bit playful in that big bones are the inner structure of a mastodon, just like this library will be the inner structure of a Mastodon client. ;)

I would personally lose the "4j" suffix, though. In my mind that's just clutter, and also hides the fact that this project is mostly Kotlin.

**Reply by andregasser** [reply]
Hehe :-) "bigbone" is fine for me. Let's be pragmatic and go with that. 👍 I did a quick google search and there was no real java library that popped up immediately. So I think we are fine in that regard.

**Reply by andregasser** [reply]
Do you have a preference regarding the package structure? To my knowledge the package structure should be based on a domain that is under our control. I think either we go with "com.github.andregasser.bigbone." or I quickly register a domain like "bigbone.io" and we put it under that domain. Actually, I prefer the latter :-)

**Reply by bocops** [reply]
A short, self-controlled URL is the best option, yes. :)

**Reply by andregasser** [reply]
What about bigbone.social? It's free. 6.87 USD/year. I can afford that

**Reply by bocops** [reply]
Sounds good. You could even host your own Mastodon instance down the line with that URL. ;)

**Reply by andregasser** [reply]
bigbone.social is purchased. Issue https://github.com/andregasser/mastodon4j/issues/31 updated.
